app/src/main/kotlin/compiler/Compiler.kt

```diff
@@ -56,16 +56,20 @@ class Compiler(val diagnostics: Diagnostics) {
 
             val programProperties = ProgramAnalyzer.analyzeProgram(astWithBuiltinFunctions, diagnostics)
 
-            val functionDetailsGenerators = FunctionDependenciesAnalyzer.createFunctionDetailsGenerators(
+            val (functionDetailsGenerators, generatorDetailsGenerators) = FunctionDependenciesAnalyzer.createFunctionDetailsGenerators(
```

Comment:
We can also add foreign generators `init`s, `resume`s and `finalize`s to foreign identifiers in line 106, similarly to functionDetailsGenerators, and remove that todo.